enhancements/split_helm_ansible_into_their_repos.md

```diff
@@ -0,0 +1,187 @@
+|  operator-sdk/hack/tests/e2e-helm.sh	|  helm-operator/hack/tests/e2e-helm.sh 	|   has the e2e test for the Helm project 	|
+
+#### Ansible
+
+| From SDK repo| To Ansible repo	| Description  	|
+|---	|---	|---	|
+|  operator-sdk/internal/cmd/ansible-operator/run 	|   ansible-operator/main.go	|   used to run the project locally	|
+| operator-sdk/internal/cmd/ansible-operator/version  	|  ansible-operator/version.go 	| define the version of the project and it used to build the bin	|
+|  operator-sdk/internal/ansible/ 	|  ansible-operator/pkg/ 	|   has the implementations used to generated the ansible based-operator image	|
+|  operator-sdk/internal/plugins/ansible/v1/ 	|  ansible-operator/pkg/plugin/v1 	|   has the implementation customized to scaffold the ansible projects 	|
+|  operator-sdk/test/e2e-ansible 	|  ansible-operator/e2e-tests 	|   has the e2e test for the ansible project 	|
+|  operator-sdk/hack/tests/e2e-ansible.sh	|  ansible-operator/hack/tests/e2e-ansible.sh 	|   has the e2e test for the ansible project 	|
+|  operator-sdk/hack/tests/e2e-ansible-molecule.sh	|  ansible-operator/hack/tests/e2e-ansible-molecule.sh 	|   has the e2e test the ansible project with molecule 	|
+|  operator-sdk/test/ansible-memcached/	|  ansible-operator/e2e-tests/ansible-memcached/ |   has the static mock data which still used by molecule tests	|
+|  operator-sdk/test/ansible/	|  ansible-operator/e2e-tests/ansible/ |   has the static mock data which still used by molecule tests	|
+
+**NOTE**: The molecule tests still via shell script, however, if they be migrated to go test stack then the mock static files will no longer be required. 
+
+### SDK Operator lib
+
+The idea of this project as described above is to centralize all code which is useful for all types projects. In this way, SDK CLI as any operator type repo can import this lib and use its features. 
+ 
+In this way follow up the SDK tree with the points that would be moved:
+
+- `internal/util/k8sutil/constants.go`: Just the `WatchNamespaceEnvVar` should be moved since it is used for both types in the `run.go` implementation and not used by the SDK CLI commands at all. 
+- `internal/plugins`: code implementation which is used to customize the scaffold of the projects and allow they work on and be integrated with the framework features (plugins phase 2). Note that ansible/helm/go plugins will not be moved to this repository.    
+- `test/internal`: the utils tests implemented in SDK can will used for any operator type and SDK CLI repo as well to implement the e2e tests.  
+
+### Usage of Helm/Ansible plugin in SDK repo 
+
+Their usage in the SDK repo will be done as it currently is made for the Go operator. See: https://github.com/operator-framework/operator-sdk/tree/master/internal/plugins/golang/v2
+
+## Test Plan
+
+The ultimate test is that e2e tests and manual testing of the SDK binary will
```

Comment:
We shouldn't be talking about manual testing here (it's a throwaway comment since we're not talking about it elsewhere), and e2e tests should be discussed only in the context of individual plugins and operators. We need to discuss how _integration_ tests will be used to determine correctness of the `operator-sdk` binary.

Reply:
I am closing this one because I better specified how I am thinking about the tests. However, please, feel free to check and give your inputs.